### PR TITLE
Build: Update build.sh

### PR DESCRIPTION
**Pull request by davidhadas**
Support podman  - no `--push` flag

### CODE REVIEW

File: hack/build.sh [hunk 1]
```diff
@@ -61,8 +61,8 @@ function build_caa_payload_image() {
 		--build-arg YQ_CHECKSUM="${YQ_CHECKSUM}" \
 		-f Dockerfile \
 		${tag_string} \
-		--push \
```
Comment (liudalibj) [inline hack/build.sh:64]: here maybe we need `--load` to replace `--push`

File: hack/build.sh [hunk 2]
```diff
@@ -106,8 +106,8 @@ function build_caa_payload_arch_specific() {
...
```
Comment (liudalibj) [inline hack/build.sh:109]: here maybe we need `--load` to replace `--push`

**Comment by davidhadas** [comment]
How can I force the tests to be redone?

File: hack/build.sh (hunk 1 above)
Comment (davidhadas) [reply hack/build.sh:64]: adding `--load` works for podman
I cannot identify a difference between running with or without it.

Comment (liudalibj) [reply hack/build.sh:64]: docker buildx build docker image inside one container instance, eg.
```
docker buildx ls
NAME/NODE        DRIVER/ENDPOINT             STATUS  BUILDKIT             PLATFORMS
multi-builder    docker-container
  multi-builder0 unix:///var/run/docker.sock running v0.12.5              linux/amd64*, linux/s390x*, linux/amd64/v2, linux/amd64/v3, linux/amd64/v4, linux/386
default *        docker
  default        default                     running v0.11.6+0a15675913b7 linux/amd64, linux/amd64/v2, linux/amd64/v3, linux/amd64/v4, linux/386
```
Without the `--load` option, the built out image only inside the container can't be list on the host machine, so the new `docker push  ${registry}/${name}` will be failed, as it try to push image from the host.

File: hack/build.sh [hunk 3]
```diff
@@ -61,8 +61,8 @@ function build_caa_payload_image() {
 		--build-arg YQ_CHECKSUM="${YQ_CHECKSUM}" \
```
Comment (davidhadas) [inline hack/build.sh:61]:
```suggestion
		--build-arg YQ_CHECKSUM="${YQ_CHECKSUM}" \
		-- load \
```

File: hack/build.sh [hunk 4]
```diff
@@ -106,8 +106,8 @@ function build_caa_payload_arch_specific() {
...
```
Comment (davidhadas) [inline hack/build.sh:106]:
```suggestion
		--build-arg YQ_CHECKSUM="${YQ_CHECKSUM}" \
		--load \
```

File: hack/build.sh (hunk 2 above)
Comment (davidhadas) [reply hack/build.sh:109]: Added --load

File: hack/build.sh (hunk 1 above)
Comment (davidhadas) [reply hack/build.sh:64]: Added --load

**Comment by davidhadas** [comment]
Replaced by #1774

**Pull request closed without merging**